src/shared/sol-conffile.c

```diff
@@ -432,28 +448,111 @@ _fill_vector(const char *path, const char *fallback_paths)
     if (_json_to_vector(json_scanner) != 0)
         goto free_for_all;
 
-    if (include || include_fallbacks)
-        _fill_vector(include, include_fallbacks);
+    if (include || include_fallbacks.len) {
+        _fill_vector(include, &include_fallbacks);
+    }
 
 free_for_all:
     sol_file_reader_close(file_reader);
     free(full_path);
     free(include);
-    free(include_fallbacks);
+
+    SOL_VECTOR_FOREACH_IDX (&include_fallbacks, slice, i) {
```

Comment:
I'm seeing more and more `sol_vector` including `sol_str_slice`, and code like this, where `free()` is called with the needed cast to `char *`. Is this maybe a cry for a `sol_str_slice_vector`?